charts/studio/templates/configmap-studio.yaml

```
@@ -56,11 +56,9 @@ data:
   REDIS_URL: "redis://{{ .Values.redis.fullnameOverride }}-master.{{ .Release.Namespace }}.svc.cluster.local:6379"
   {{- end }}
 
-  {{- if .Values.global.scmProviders.github.apiUrl }}
-  GITHUB_API_URL: {{ .Values.global.scmProviders.github.apiUrl | quote }}
-  {{- end }}
   {{- if .Values.global.scmProviders.github.url }}
   GITHUB_URL: {{ .Values.global.scmProviders.github.url | quote }}
+  GITHUB_API_URL: "{{ .Values.global.scmProviders.github.url | trimSuffix "/" }}/api/v3"
```

https://github.com is under https//api.github.com. 

Please add a condition which sets `.Values.global.scmProviders.github.url` as default if `GITHUB_API_URL` is not defined

---

I'm not sure I follow the second part, but I understand your concern.

A user might set `github.url` to `https://github.com` by mistake. This would of course break the API, as it is on the `api.github.com` domain as you pointed out.

How about checking if `github.url` does not contain `github.com` before setting `GITHUB_API_URL`? This way, we know we're dealing with a Github Enterprise instance

```suggestion
{{- if not contains "github.com" .Values.global.scmProviders.github.url }}
  GITHUB_API_URL: "{{ .Values.global.scmProviders.github.url | trimSuffix "/" }}/api/v3"
{{- end }}
```

---

Maybe it's safest to keep url and apiUrl independent for now but set api url like so, to ensure suffix:
```
"{{ .Values.global.scmProviders.github.apUrl | trimSuffix "/" | trimSuffix "/api/v3" }}/api/v3"
```

---

> How about checking if `github.url` does not contain `github.com` before setting `GITHUB_API_URL`? This way, we know we're dealing with a Github Enterprise instance

Good Idea

@jesper7 where did you found this requirement about `api/v3` path? I though you are talking about GitLab ^^

---

> But I think it's just not always correct, right? for github.com the api is https://api.github.com and NOT https://api.github.com/api/v3

That's right. But suppose that a user wants to use SaaS GitHub; they shouldn't override `github.url` because `https://github.com` is already the default in the Docker images. 
If they do override it by accident, we can use my previous proposal to ensure that it won't override `GITHUB_API_URL` either (the default is `https://api.github.com` in the Docker images)

I don't think we should make `GITHUB_API_URL` configurable because it will always be `<domain>/api/v3` for GitHub Enterprise.

> @jesper7 where did you found this requirement about api/v3 path? I though you are talking about GitLab ^^

It was discovered during a troubleshooting session with a customer and afterward confirmed by the backend engineers.

---

@jesper7  - IIUC you are suggesting that not only would setting `GITHUB_URL` and `GITHUB_API_URL` be unnecessary for github SaaS, but it would be wrong to do so and would fail / break even if you have the correct addresses.
I don't think that's an expected behavior. Needing to set fields to different values in different scenarios is one thing, but remembering / keeping in mind that in some scenario (gh saas) you're just not allowed to provide some fields (even though you provide correct values there) just because of some internal voodoo is not intuitive and could lead to frustration.
In other words - I think it's weird to have defaults (even if internal) that wouldn't pass through input validation. I think it's best to keep "symmetry" and in this case and just properly document (comments in values.yaml) for pitfalls
But this is just a personal take, so take it or leave it :) no worries

---

> but it would be wrong to do so and would fail / break even if you have the correct addresses.

It will work if you provide the correct values for SaaS GitHub (`GITHUB_URL=https://github.com` and `GITHUB_API_URL=https://api.github.com`); however, these are already set by default in the Docker images, so repeating them in the Helm chart values.yaml is redundant.

I do agree with your other points, though. How about entirely getting rid of the `GITHUB_API_URL` environment variable and simply setting it automatically in the backend? I can submit a draft PR to give an example.